### DISCUSSION TITLE
Need to deploy Standard Logic app workflows from one environment to another using ARM templates

### DISCUSSION THREAD

**Discussion by Panchadurga** [Azure/logicapps#766] [Q&A]
Can anyone please suggest on how to deploy Standard Logic app workflows from one environment to another using ARM templates?

**Comment by mark-abrams** [comment]
Hi, Standard Logic App workflows are not deployed using ARM templates. Only Consumption logic app workflows are deployed in this way.

With Standard Logic Apps, the Logic App infrastructure is separate from the workflows that run in the Logic App. So the Logic App itself is deployed using ARM templates, or Bicep or Terraform for example. The workflows and related connections and parameters are deployed separately as part of a DevOps process, check out [this article](https://learn.microsoft.com/en-us/azure/logic-apps/set-up-devops-deployment-single-tenant-azure-logic-apps).

This deployment approach is the same as Function Apps deployments, since Standard Logic Apps are just a specialised type of Function Apps.

**Reply by Panchadurga** [reply]
Thanks for your quick response.

Is there any other way to deploy standard logic app workflows without using Azure DevOps Pipelines?

**Comment by mark-abrams** [comment]
You could use the ZIP deployment method, using Azure CLI or PowerShell to do this.  This is taken from the [link I shared earlier](https://learn.microsoft.com/en-us/azure/logic-apps/set-up-devops-deployment-single-tenant-azure-logic-apps?tabs=azure-cli):

![image](https://user-images.githubusercontent.com/66829458/234317352-75fa0819-17c4-487c-82c5-28dab350d0ce.png)

But the most common way to deploy Standard Logic App workflows is using a DevOps pipeline in your preferred DevOps platform - GitHub, Azure or whatever. Which ever approach you take, they are all essentially doing the same thing, deploying a set of artifacts (workflows, connection config, parameter config, local.settings files) to an Azure Logic App.

**Reply by Panchadurga** [reply]
Thank you so much.

**Comment by Panchadurga** [comment]
Hi @mark-abrams, 

The deployment of Logic Apps Standard via a CI/CD pipeline requires a different approach than that of Logic Apps Consumption(ISE). Because Logic Apps Standard is now a number of workflows in a single component.
For Example:
ARM (template and parameter) files were used in ISE to automate the deployment of logic apps from one environment to another environment. However, parameter files are only available at the global (logic app) level in Standard, not at the workflow level. 

Can you please confirm that only **zip deployment** is possible using CLI commands, or is it possible to deploy workflows alone, and how parameter files will be maintained in Standard Logic App?

Thanks
